akka-actor/src/main/scala/akka/pattern/BackoffOptions.scala

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.pattern
+
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+import akka.actor.{ Props, OneForOneStrategy, SupervisorStrategy }
+import akka.actor.SupervisorStrategy.{ Decider, JDecider }
+
+/**
+ * Builds back-off options for creating a back-off supervisor.
+ * You can pass `BackoffOptions` to [[akka.pattern.BackoffSupervisor#props]].
+ * An example of creating back-off options:
+ * {{{
+ * BackoffOptions.onFailure(childProps, childName, minBackoff, maxBackoff, randomFactor)
+ *               .withManualReset
+ *               .withSupervisorStrategy(
+ *                 OneforOneStrategy(){
+ *                    case e: GivingUpException => Stop
+ *                    case e: RetryableException => Restart
+ *                 }
+ *               )
+ *
+ * }}}
+ */
+object BackoffOptions {
+  /**
+   * Back-off options for creating a back-off supervisor actor that expects a child actor to restart on failure.
+   *
+   * This explicit supervisor behaves similarly to the normal implicit supervision where
+   * if an actor throws an exception, the decider on the supervisor will decide when to
+   * `Stop`, `Restart`, `Escalate`, `Resume` the child actor.
+   *
+   * When the `Restart` directive is specified, the supervisor will delay the restart
+   * using an exponential back off strategy (bounded by minBackoff and maxBackoff).
+   *
+   * This supervisor is intended to be transparent to both the child actor and external actors.
+   * Where external actors can send messages to the supervisor as if it was the child and the
+   * messages will be forwarded. And when the child is `Terminated`, the supervisor is also
+   * `Terminated`.
+   * Transparent to the child means that the child does not have to be aware that it is being
+   * supervised specifically by this actor. Just like it does
+   * not need to know when it is being supervised by the usual implicit supervisors.
+   * The only caveat is that the `ActorRef` of the child is not stable, so any user storing the
+   * `sender()` `ActorRef` from the child response may eventually not be able to communicate with
+   * the stored `ActorRef`. In general all messages to the child should be directed through this actor.
+   *
+   * An example of where this supervisor might be used is when you may have an actor that is
+   * responsible for continuously polling on a server for some resource that sometimes may be down.
+   * Instead of hammering the server continuously when the resource is unavailable, the actor will
+   * be restarted with an exponentially increasing back off until the resource is available again.
+   *
+   * '''***
+   * This supervisor should not be used with `Akka Persistence` child actors.
+   * `Akka Persistence` actors, currently, shutdown unconditionally on `persistFailure()`s rather
```

Comment:
the word "currently" is not needed, we have no plans on changing and software is always changing